akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -135,6 +135,7 @@ abstract class SerializerWithStringManifest extends Serializer {
    * and message is dropped. Other exceptions will tear down the TCP connection
    * because it can be an indication of corrupt bytes from the underlying transport.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
Right, this may break source compatibility, but not binary (runtime compatibility, since checked exceptions are only a compile time fake). I think it's alright to break source compatibility because this is the right thing to declare, since it may anyway throw that and currently there is no good way to catch that from java.